tools/CheckFiles.sh

```diff
@@ -107,38 +112,82 @@ find ./ -type f                                                              \
      ! -name "*.patch"                                                       \
      ! -name "*LICENSE.*"                                                    \
      ! -name "*.clang-format"                                                \
+     ! -path "./build*"                                                      \
     | xargs grep -L "^.*Distributed under the MIT License"`
 if [[ $files_without_license != "" ]]; then
     echo "This script can be run locally from any source dir using:"
     echo "SPECTRE_ROOT/tools/CheckFiles.sh"
     echo "Did not find a license in these files:"
     echo "$files_without_license"
-    exit 1
+    found_error=1
 fi
 
 ###############################################################################
 # Check for tests using Catch's TEST_CASE instead of SPECTRE_TEST_CASE
 found_test_case=`
-find ./ -type f -name '*.[ch]pp' \
+find ./ -type f -name '*.[ch]pp' -not -path './build*' \
     | xargs grep --with-filename -n "^TEST_CASE"`
 if [[ $found_test_case != "" ]]; then
     echo "This script can be run locally from any source dir using:"
     echo "SPECTRE_ROOT/tools/CheckFiles.sh"
     echo "Found occurrences of TEST_CASE, must use SPECTRE_TEST_CASE:"
     echo "$found_test_case"
-    exit 1
+    found_error=1
 fi
 
 ###############################################################################
 # Check for tests using Catch's Approx, which has a very loose tolerance
 found_bad_approx=`
-find ./ -type f -name '*.[ch]pp' \
+find ./ -type f -name '*.[ch]pp' -not -path './build*' \
     | xargs grep --with-filename -n "Approx("`
 if [[ $found_bad_approx != "" ]]; then
     echo "This script can be run locally from any source dir using:"
     echo "SPECTRE_ROOT/tools/CheckFiles.sh"
     printf "Found occurrences of Approx, must use approx from " \
            "SPECTRE_ROOT/tests/Unit/TestHelpers.hpp instead:\n"
     echo "$found_bad_approx"
+    found_error=1
+fi
+
+###############################################################################
+# Check for Doxygen group inclusions on the starting line
+found_incorrect_ingroup_syntax=`
+find ./ -type f -name '*.[ch]pp' -not -path './build*' \
+    | xargs grep --with-filename -n '/\*\![ ]*\\\\ingroup'`
```

Comment:
Would it make sense to just ban any "/*! \", rather than checking for particular commands?

Reply:
That was supposed to have a backslash in it, but it seems to have been eaten by GitHub.

Reply:
I'd be happy with banning "/*!.+" (if grep can do a line by line search and does not match newlines with `.`) The `/*!` comments are useful (needed?) for multiline math and I have developed a personal preference for them.

Reply:
I intended to suggest banning slash-star-bang-space*-backslash, but GitHub stripped the backslash (and I forgot the *, or + if you prefer).  Plain slash-star-bang should certainly be allowed.

Reply:
Okay, we are in agreement then, I will make the change.